<a href="https://colab.research.google.com/github/Nurochman79/DataScience/blob/main/HR_Data_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

Membaca data format csv

In [ ]:
hr_data=pd.read_csv('/content/HR_comma_sep (1).csv')

Melihat format detail data: nama variabel, jumah data, dan type data. Jika jumalah data tidak sama maka perlu dihandling

In [ ]:
hr_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   satisfaction_level     14999 non-null  float64
 1   last_evaluation        14999 non-null  float64
 2   number_project         14999 non-null  int64  
 3   average_montly_hours   14999 non-null  int64  
 4   time_spend_company     14999 non-null  int64  
 5   Work_accident          14999 non-null  int64  
 6   left                   14999 non-null  int64  
 7   promotion_last_5years  14999 non-null  int64  
 8   sales                  14999 non-null  object 
 9   salary                 14999 non-null  object 
dtypes: float64(2), int64(6), object(2)
memory usage: 1.1+ MB


Dari data di atas jumlah data sama sehingga diduga data tadak ada missing. Untuk itu perlu dipastikan seperti di bawah.

In [ ]:
hr_data.isnull().sum()

satisfaction_level       0
last_evaluation          0
number_project           0
average_montly_hours     0
time_spend_company       0
Work_accident            0
left                     0
promotion_last_5years    0
sales                    0
salary                   0
dtype: int64

In [ ]:
hr_data.isna().sum()

satisfaction_level       0
last_evaluation          0
number_project           0
average_montly_hours     0
time_spend_company       0
Work_accident            0
left                     0
promotion_last_5years    0
sales                    0
salary                   0
dtype: int64

Dari pengecekan di atas tidak terdapat nilai data null ataupun NA sehingga data lengkap dan konsisten.

In [ ]:
hr_data['left'].value_counts()

0    11428
1     3571
Name: left, dtype: int64

Berdasarkan data di atas ada 3571 karyawan yang resign